types/tx_msg.go

```diff
@@ -79,3 +83,22 @@ type TxEncoder func(tx Tx) ([]byte, error)
 func MsgTypeURL(msg Msg) string {
 	return "/" + proto.MessageName(msg)
 }
+
+// GetMsgFromTypeURL returns a `sdk.Msg` message type from a type URL
+func GetMsgFromTypeURL(cdc codec.Codec, input string) (Msg, error) {
```

Comment:
I'm not understanding how this works. Can we add a unit test for it please?

Reply:
Sure, adding!

Reply:
I do understand how it works now though 👍